tests/github/interactive_api_director/experiments/pytorch_kvasir_unet/envoy/kvasir_shard_descriptor.py

```diff
@@ -0,0 +1,142 @@
+# Copyright (C) 2020-2021 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+"""Kvasir shard descriptor."""
+
+
+import os
+from pathlib import Path
+from typing import List
+
+import numpy as np
+from PIL import Image
+
+from openfl.interface.interactive_api.shard_descriptor import ShardDescriptor
+from openfl.utilities import validate_file_hash
+
+
+class KvasirShardDescriptor(ShardDescriptor):
+    """Shard descriptor class."""
+
+    def __init__(self, data_folder: str = 'kvasir_data',
+                 rank=1,
+                 worldsize=1,
+                 enforce_image_hw: List[int] = None) -> None:
+        """Initialize KvasirShardDescriptor."""
+        super().__init__()
+        
+        self.data_folder = Path.cwd() / data_folder
+        self.download_data(self.data_folder)
+
+        # Settings for resizing data
+        self.enforce_image_hw = None
+        if enforce_image_hw is not None:
+            self.enforce_image_hw = tuple(enforce_image_hw)
```

Comment:
To allow using `Tuple` here we need to parse YAML collections as tuples. But It requires a specific `!!python/tuple` tag that is not supported by `yaml.safe_load` function. All collections are parsed as lists and they need to be converted to tuples inside `ShardDescriptor` class since the usage of `yaml.safe_load` is strictly required.

Reply:
Maybe we can convert it to Tuple before we are passing the arguments to constructor?

Reply:
We probably could if we created `ShardDescriptor` instance manually only. But when we spawn envoy via `fx envoy start`, it reads Shard Descriptor YAML file, calls the constructor of `template` class, and passes `settings` as keyword arguments. In this case, the constructor will always receive collections as lists. We can not access intermediate steps between parsing the config file and calling the constructor because it happens inside the library code.